admin/controllers/check-form.js

```diff
@@ -123,88 +124,111 @@ const saveCheckForm = async (req, res, next) => {
+  uploadError.message = 'A valid CSV file was not uploaded'
+  uploadError.errors = {}
+  uploadError.errors['csvFile'] = new Error(uploadError.message)
+
+  // Either it actually wasn't uploaded, or it failed one the busboy checks: e.g.
+  // * mime-type needs to be text/csv (.csv)
+  // * uploaded from the wrong path
+  // * file size exceeded?
+  if (!uploadFile) {
+    return res.render('test-developer/upload-new-form', {
+      error: uploadError,
+      breadcrumbs: req.breadcrumbs()
+    })
+  }
 
+  // Convert single uploaded file to an array
+  if (!Array.isArray(uploadFile)) {
+    uploadFile = [uploadFile]
+  }
+
+  // If a non-csv file was uploaded, fail with error
+  if (!R.all((file) => path.extname(file.filename.toLowerCase()) === '.csv', uploadFile)) {
     return res.render('test-developer/upload-new-form', {
       error: uploadError,
       breadcrumbs: req.breadcrumbs()
     })
   }
 
   /**
-   * uploadfile is an object:
+   * uploadFile is an array of objects like:
    *
-   { uuid: 'ff6c17d9-84d0-4a9b-a3c4-3f94a6ccdc40',
+   [ { uuid: 'ff6c17d9-84d0-4a9b-a3c4-3f94a6ccdc40',
      field: 'uploadFile',
      file: 'data/files/ff6c17d9-84d0-4a9b-a3c4-3f94a6ccdc40/uploadFile/form-1.csv',
      filename: 'form-1.csv',
      encoding: '7bit',
      mimetype: 'text/csv',
      truncated: false,
-     done: true } }
+     done: true } ]
    */
 
-  absFile = path.join(__dirname, '/../', uploadFile.file)
-  deleteDir = path.dirname(path.dirname(absFile))
+  let checkForms = []
+  for (let i = 0; i < uploadFile.length; i++) {
+    let checkForm = {}
+    absFile = path.join(__dirname, '/../', uploadFile[i].file)
+    deleteDir = path.dirname(path.dirname(absFile))
 
-  try {
-    await checkFormService.populateFromFile(checkForm, absFile)
-  } catch (error) {
-    fs.remove(deleteDir, err => {
-      if (err) winston.error(err.message)
-    })
-    return res.render('test-developer/upload-new-form', {
-      error: new Error('There is a problem with the form content'),
-      breadcrumbs: req.breadcrumbs()
-    })
-  }
+    try {
+      await checkFormService.populateFromFile(checkForm, absFile)
+    } catch (error) {
+      fs.remove(deleteDir, err => {
+        if (err) winston.error(err.message)
+      })
+      return res.render('test-developer/upload-new-form', {
+        error: new Error(`There is a problem with the form content - ${uploadFile[i].filename}`),
+        breadcrumbs: req.breadcrumbs()
+      })
+    }
 
-  try {
-    fileName = await checkFormService.buildFormName(uploadFile.filename)
-    if (!fileName) {
-      req.flash('error', `Select a file with no more than 128 characters in name`)
-      return res.redirect('/test-developer/upload-new-form')
+    try {
+      fileName = await checkFormService.buildFormName(uploadFile[i].filename)
+      if (!fileName) {
+        req.flash('error', `Select a file with no more than 128 characters in name - ${uploadFile[i].filename.slice(0, -4)}`)
```

Comment:
I noticed last night that the DB only accepts 60 chars.   We should have a consistent length.

Reply:
@danwilliams aside from this comment around the length from @jon-shipley the PR looks good. Is this going to be changed in this PR?